mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,370 @@
+        y_pred = self.predict(X)
+        n_outputs = y_pred.shape[-1]
+        y_pred = y_pred.reshape(y.shape, order='F')
+        y_pred = y_pred[self.keep_samples_]
+        y = y[self.keep_samples_]
+
+        # Re-vectorize and call scorer
+        y = y.reshape([-1, n_outputs], order='F')
+        y_pred = y_pred.reshape([-1, n_outputs], order='F')
+        scores = self.scorer_(y, y_pred, multioutput='raw_values')
+        return scores
+
+    def _check_dimensions(self, X, y, predict=False):
+        if X.ndim == 1:
+            raise ValueError('X must be shape (n_times[, n_epochs],'
+                             ' n_features)')
+        elif X.ndim == 2:
+            # Ensure we have a 3D input by adding singleton epochs dimension
+            X = X[:, np.newaxis, :]
+            if y is None:
+                pass
+            elif y.ndim == 1:
+                y = y[:, np.newaxis, np.newaxis]  # Add epochs and outputs dim
+            elif y.ndim == 2:
+                y = y[:, np.newaxis]  # Only add epochs dim
+            else:
+                raise ValueError('y must be shape (n_times[, n_epochs],'
+                                 ' n_outputs')
+        elif X.ndim == 3:
+            if y is None:
+                pass
+            elif y.ndim == 2:
+                y = y[:, :, np.newaxis]  # Add an outputs dim
+            elif y.ndim != 3:
+                raise ValueError('If X has 3 dimensions, '
+                                 'y must be at least 2 dimensions')
+        else:
+            raise ValueError('X must be of shape '
+                             '(n_times[, n_epochs], n_features)')
+        if y is None:
+            # If y is None, then we don't need any more checks
+            return X, y
+
+        if X.shape[0] != y.shape[0]:
+            raise ValueError('X any y do not have the same n_times\n'
+                             '%s != %s' % (X.shape[0], y.shape[0]))
+        if X.shape[1] != y.shape[1]:
+            raise ValueError('X any y do not have the same n_epochs\n'
+                             '%s != %s' % (X.shape[1], y.shape[1]))
+        if predict is True:
+            if y.shape[-1] != len(self.estimator_.coef_):
+                raise ValueError('Number of outputs does not match'
+                                 ' estimator coefficients dimensions')
+        return X, y
+
+
+def _delay_time_series(X, tmin, tmax, sfreq, newaxis=0, axis=0):
+    """Return a time-lagged input time series.
+
+    Parameters
+    ----------
+    X : array, shape (n_times[, n_epochs], n_features)
+        The time series to delay.
+    tmin : int | float
+        The starting lag. Negative values correspond to times in the past.
+    tmax : int | float
+        The ending lag. Positive values correspond to times in the future.
+        Must be >= tmin.
+    sfreq : int | float
+        The sampling frequency of the series. Defaults to 1.0.
+    newaxis : int
```

Comment:
it's a private function - could be useful to create the delay axis on an arbitrary axis